governance/program/src/state/token_owner_record.rs

```diff
@@ -0,0 +1,168 @@
+    pub token_deposit_amount: u64,
+
+    /// A single account that is allowed to operate governance with the deposited governing tokens
+    /// It's delegated to by the governing token owner or current governance_delegate
+    pub governance_delegate: Option<Pubkey>,
+
+    /// The number of active votes cast by TokenOwner
+    pub active_votes_count: u8,
+
+    /// The total number of votes cast by the TokenOwner
+    pub total_votes_count: u8,
+}
+
+impl AccountMaxSize for TokenOwnerRecord {
+    fn get_max_size(&self) -> Option<usize> {
+        Some(140)
+    }
+}
+
+impl IsInitialized for TokenOwnerRecord {
+    fn is_initialized(&self) -> bool {
+        self.account_type == GovernanceAccountType::TokenOwnerRecord
+    }
+}
+
+/// Returns TokenOwnerRecord PDA address
+pub fn get_token_owner_record_address(
+    realm: &Pubkey,
+    governing_token_mint: &Pubkey,
+    governing_token_owner: &Pubkey,
+) -> Pubkey {
+    Pubkey::find_program_address(
+        &get_token_owner_record_address_seeds(realm, governing_token_mint, governing_token_owner),
+        &id(),
+    )
+    .0
+}
+
+/// Returns TokenOwnerRecord PDA seeds
+pub fn get_token_owner_record_address_seeds<'a>(
+    realm: &'a Pubkey,
+    governing_token_mint: &'a Pubkey,
+    governing_token_owner: &'a Pubkey,
+) -> [&'a [u8]; 4] {
+    [
+        PROGRAM_AUTHORITY_SEED,
+        realm.as_ref(),
+        governing_token_mint.as_ref(),
+        governing_token_owner.as_ref(),
+    ]
+}
+
+/// Deserializes TokenOwnerRecord account and checks owner program
+pub fn deserialize_token_owner_record_raw(
+    token_owner_record_info: &AccountInfo,
+) -> Result<TokenOwnerRecord, ProgramError> {
+    deserialize_account::<TokenOwnerRecord>(token_owner_record_info, &id())
+}
+
+/// Deserializes TokenOwnerRecord account and checks its PDA against the provided seeds
+pub fn deserialize_token_owner_record(
+    token_owner_record_info: &AccountInfo,
+    token_owner_record_seeds: &[&[u8]],
+) -> Result<TokenOwnerRecord, ProgramError> {
+    let (token_owner_record_address, _) =
+        Pubkey::find_program_address(token_owner_record_seeds, &id());
+
+    if token_owner_record_address != *token_owner_record_info.key {
+        return Err(GovernanceError::InvalidTokenOwnerRecordAccountAddress.into());
+    }
+
+    deserialize_token_owner_record_raw(token_owner_record_info)
+}
+
+/// Deserializes TokenOwnerRecord account and checks that its PDA matches the given realm and governing mint
+pub fn deserialize_token_owner_record_for_realm_and_governing_mint(
+    token_owner_record_info: &AccountInfo,
+    realm: &Pubkey,
+    governing_token_mint: &Pubkey,
+) -> Result<TokenOwnerRecord, ProgramError> {
+    let token_owner_record_data = deserialize_token_owner_record_raw(token_owner_record_info)?;
+    let token_owner_record_address = get_token_owner_record_address(
+        realm,
+        &governing_token_mint,
+        &token_owner_record_data.token_owner,
```

Comment:
What I'm after is to verify the `token_owner_record_info` is for the given `realm` and `governing_token_mint`. The actual `token_owner` is not known upfront (and doesn't have to be) and this is why it's taken from the account after deserialisation. 
However that code was sort of legacy when we didn't have the `governing_token_mint` on `token_owner_record_info`. Since the mint is there now I changed the code to verify the fields directly instead of PDA